### PR TITLE
Feature/improvements and bugfixes

### PR DESCRIPTION
**Pull request by sinadarbouy**
### Description:
This feature branch was initially developed to add Pod Disruption Budgets (PDB) and readiness checks. During this process, some challenges were encountered. To address these and further enhance the helm chart, this PR includes a series of improvements and bug fixes:

### Features
1. **Pod Disruption Budgets:**
Added Pod Disruption Budgets to maintain application availability during voluntary disruptions.
2. **Readiness Probe:**
Introduced a readinessProbe to enhance the deployment's resilience and ensure only healthy pods receive traffic.
3. **Gatewayd Configuration `gatewayd.yaml`:**
Added a `gatewayd.yaml` ConfigMap to allow users to overwrite the `gatewayd.yaml` file.
This enables updating the client endpoint and other necessary parameters for better management.
Updated configuration for gatewayd to allow updating the client endpoint and other necessary parameters for better management.
4. **README Updates:**
Documented the addition of Pod Disruption Budgets.
Updated the gatewayd configuration section to reflect the recent changes.

### Bugfix or Improvement 
1. **Update App Version:**
Updated appVersion in chart.yaml to the latest version.

2. **Plugin Install Command:**
Added a command to install plugins
> If I understand correctly, we need to run the `plugin install` command before the `gatewayd run` command to get binaries for the plugins.
### Refactoring:
1. **Configuration Management:**
Moved multi-line string configurations to a dedicated files folder to simplify linting and management of configuration files.
2. **ConfigMap Renaming:**
Renamed `gatewayd-gatewayd-plugin-config` ConfigMap to remove redundancy, simplifying the name to `gatewayd-plugin-config`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,13 +77,21 @@ helm install gatewayd-release -f values.yaml ./
...
 | `serviceAccount.automount`                  | Determines whether the ServiceAccount token should be automatically mounted to the pods. This is set as the `automountServiceAccountToken` field in the ServiceAccount resource. |                             |
 
+### Pod Disruption Budgets Configuration
+
+| Parameter                                   | Description                                                                                           | Default Value               |
+|---------------------------------------------|-------------------------------------------------------------------------------------------------------|-----------------------------|
+| `podDisruptionBudget.minAvailable`                   | which is a description of the number of pods from that set that must still be available after the eviction, even in the absence of the evicted pod.                      |                             |
+| `podDisruptionBudget.maxUnavailable`                   | which is a description of the number of pods from that set that can be unavailable after the eviction. It can be either an absolute number or a percentage.                    |                             |
```
Comment (mostafa) [inline README.md]: Fix descriptions.

**Review by mostafa — COMMENTED** [review]
Overall LGTM! I left a comment.

File: README.md (hunk 1 above)
Comment (sinadarbouy) [reply README.md]: Fixed

- Set default value.
- Refactored descriptions.

**Review by mostafa — APPROVED** [review]

**Comment by mostafa** [comment]
@sinadarbouy Tagged a new release: https://github.com/gatewayd-io/helm-charts/releases/tag/v0.1.2

**Pull request merged**